# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1363847469 14400
#      Thu Mar 21 02:31:09 2013 -0400
# Node ID d8babb249f65699f11328ccbb992736053d85c2d
# Parent  d4b6ad43bc8772acc7adb04d3900840b3c02a928
scroll to bottom of history list directly

* history-dockwidget.cc
(history_dock_widget::update_history_callback): Delete static variable
scroll_window.  Call scrollToBottom if the list changes instead of
delaying to the next callback.

diff --git a/libgui/src/history-dockwidget.cc b/libgui/src/history-dockwidget.cc
--- a/libgui/src/history-dockwidget.cc
+++ b/libgui/src/history-dockwidget.cc
@@ -129,18 +129,16 @@ void
 history_dock_widget::reset_model ()
 {
   _history_model->setStringList (QStringList ());
 }
 
 void
 history_dock_widget::update_history_callback (void)
 {
-  static bool scroll_window = false;
-
   // Determine the client's (our) history length and the one of the server.
   int clientHistoryLength = _history_model->rowCount ();
   int serverHistoryLength = command_history::length ();
 
   // If were behind the server, iterate through all new entries and add
   // them to our history.
   if (clientHistoryLength < serverHistoryLength)
     {
@@ -151,23 +149,11 @@ history_dock_widget::update_history_call
       for (int i = clientHistoryLength; i < serverHistoryLength; i++)
         {
           std::string entry = command_history::get_entry (i);
 
           _history_model->setData (_history_model->index (i),
                       QString::fromUtf8 (entry.data (), entry.size ()));
         }
 
-      // FIXME -- does this behavior make sense?  Calling
-      // _history_list_view->scrollToBottom () here doesn't seem to
-      // have any effect.  Instead, we need to request that action
-      // and wait until the next event occurs in which no items
-      // are added to the history list.
-
-      scroll_window = true;
-    }
-  else if (scroll_window)
-    {
-      scroll_window = false;
-
       _history_list_view->scrollToBottom ();
     }
 }
